python_modules/dagster/dagster/_core/definitions/asset_checks.py

```diff
@@ -29,15 +31,29 @@
 if TYPE_CHECKING:
     from dagster._core.definitions.assets import AssetsDefinition
 
+TPropsType = TypeVar("TPropsType", bound=NamedTuple)
+TDefType = TypeVar("TDefType")
 
-@experimental
-class AssetChecksDefinitionInputOutputProps(NamedTuple):
+
+class IDefinitionPropsClassMatchesInit(Generic[TPropsType, TDefType], ABC):
+    @staticmethod
+    def from_props(props: TPropsType) -> TDefType: ...
```

Comment:
this appears unused - should we instead capture [1] with a method on this class?